File: python_75/two/buy_sell_stocks.py
# input = [7, 5, 3, 1, 8, 4, 9]
# output = 8

nums = [7, 5, 3, 1, 8, 4, 9,22]

maxProfit = 0
min = nums[0]

for cur in nums:
  if cur < min:
    min = cur
  profit = cur - min
  maxProfit = max(maxProfit, profit)

print(maxProfit)

File: spring-data-redis-example/src/main/java/com/sunil/redis/ds/buySellStocks.java
package com.sunil.redis.ds;

public class buySellStocks {

    public static void main(String[] args) {
        int[] nums = {4, 5, 1, 3, 7, 9, 2};
        System.out.println(buySell(nums));
    }

    public static int buySell(int[] nums) {

        int maxProfit = 0;
        int min = nums[0];

        for (int num : nums) {
            if (num < min)
                min = num;
            int curProfit = num - min;
            maxProfit = Math.max(curProfit, maxProfit);
        }
        return maxProfit;
    }
}
